quinn-proto/src/endpoint.rs

```diff
@@ -521,17 +476,48 @@ impl Endpoint {
                     return Some(DatagramEvent::Response(self.initial_close(
                         version,
                         addresses,
-                        crypto,
+                        &crypto,
                         &src_cid,
                         TransportError::INVALID_TOKEN(""),
                         buf,
                     )));
                 }
             }
-        } else {
-            (None, dst_cid)
         };
 
+        Some(DatagramEvent::NewConnection(IncomingConnection {
+            addresses,
+            ecn,
+            packet,
+            rest,
+            crypto,
+            src_cid,
+            dst_cid,
+            packet_number,
+            version,
+            retry_src_cid,
+            orig_dst_cid,
+        }))
+    }
+
+    /// Attempt to accept this incoming connection (an error may still occur)
+    pub fn accept(
+        &mut self,
+        incoming: IncomingConnection,
+        now: Instant,
+        buf: &mut BytesMut,
+    ) -> Result<(ConnectionHandle, Connection), Option<Transmit>> {
+        self.check_connection_limit(
+            incoming.version,
+            incoming.addresses,
+            &incoming.crypto,
+            &incoming.src_cid,
+            buf,
+        )
+        .map_err(Some)?;
```

Comment:
This `map_err` is not required, because `Option<Transmit>` satisfies `From<Transmit>`.

Reply:
Obviated by changing this method's error type to `(ConnectionError, Option<Transmit>)` which means it does require a non-trivial `map_err`.